libobs-opengl/gl-shaderparser.c

```diff
@@ -741,7 +752,7 @@ static bool gl_shader_buildstring(struct gl_shader_parser *glsp)
 		return false;
 	}
 
-	dstr_copy(&glsp->gl_string, "#version 330\n\n");
+	dstr_copy(&glsp->gl_string, "#version 460\n\n");
```

Comment:
No associated changes with our required openGL version? I presume it takes an OpenGL 4.6 context to compile glsl 4.6?

Reply:
> No associated changes with our required openGL version? I presume it takes an OpenGL 4.6 context to compile glsl 4.6?

I know that it works fine when testing on Windows and Linux, and I see no warnings about shader version and context mismatch. Looking more into what is appropriate...

Reply:
I do not have exact documentation confirming this precisely, but it seems possible for an older GL context to compile a shader with a newer version: https://stackoverflow.com/questions/35354496/why-does-my-version-330-shader-run-on-older-opengl

It could be more  appropriate to force the required GL version to be 4.6. It appears a newer version of the GLAD library would be needed, because the one currently in the build tree seems to only have awareness of 4.3.

I think ideally a 4.6 version will be requested for GLSL programs that need results, and an older 3.3 could be used on everything else. This would be nice because users with older hardware can continue using OBS, and will only run into problems when newer features, such as results, are requested by plugins. 

I have tried, in the past, to implement atomic counters using an ARB extension to 3.30 instead of upping the version to 4.6. This, unfortunately, has somehow ended up breaking shaders for the existing effects, and so has created more problems than it has fixed.

I really hope that upping the GL version will not be a deal breaker. It could be good to observe the behavior on older hardware that only supports 3.30, but I do not exactly have access to that. The closest I have is a linux machine that runs mesa drivers and they only support version 4.3. I will have to confirm this to be sure, but what I remember happens is this: When running OBS (with our changes) everything that does not require 4.6 works (even when shaders request version 460). But then when features needed for the atomic counters are actually used in the shader the effect will break, and only that effect. Other effects and the rest of OBS would continue working fine.

Reply:
macOS deprecated OpenGL long ago and supports 4.10 max. So without a Metal renderer this would've been a non-starter for us.